core/src/main/java/com/linecorp/armeria/server/ShutdownSupport.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static com.linecorp.armeria.server.Server.logger;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.common.util.UnmodifiableFuture;
+import com.linecorp.armeria.server.logging.AccessLogWriter;
+
+interface ShutdownSupport extends Supplier<CompletableFuture<Void>> {
```

Comment:
`Supplier<CompletableFuture<Void>>` seems to make the purpose of this interface difficult to read.
How about adding a dedicated method like `CompletableFuture<Void> shutdown()` for readiblity?
`shutdown()` could deliver a more proper behavior than `Supplier.get()`.

Reply:
That's a good suggestion. 👍